src/EditorFeatures/CSharp/Wrapping/SeparatedSyntaxList/CSharpParameterWrapper.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Editor.Wrapping.SeparatedSyntaxList
+{
+    internal partial class CSharpParameterWrapper
+        : AbstractCSharpSeparatedSyntaxListWrapper<BaseParameterListSyntax, ParameterSyntax>
+    {
+        protected override string ListName => FeaturesResources.parameter_list;
+        protected override string ItemNamePlural => FeaturesResources.parameters;
+        protected override string ItemNameSingular => FeaturesResources.parameter;
+
+        protected override SeparatedSyntaxList<ParameterSyntax> GetListItems(BaseParameterListSyntax listSyntax)
+            => listSyntax.Parameters;
+
+        protected override BaseParameterListSyntax GetApplicableList(SyntaxNode node)
+            => CSharpSyntaxGenerator.GetParameterList(node);
+
+        protected override bool PositionIsApplicable(
+            SyntaxNode root, int position, SyntaxNode declaration, BaseParameterListSyntax listSyntax)
+        {
+            // CSharpSyntaxGenerator.GetParameterList synthesizes a parameter list for simple-lambdas.
+            // In that case, we're not applicable in that list.
+            if (declaration.Kind() == SyntaxKind.SimpleLambdaExpression)
+            {
+                return false;
+            }
+
+            var generator = CSharpSyntaxGenerator.Instance;
+            var attributes = generator.GetAttributes(declaration);
```

Comment:
why get attributes are in generator?

Reply:
So that people can use the generator in a language agnostic fashion.  The generator is full of these helpers.